### ISSUE TITLE
Centos 8 : error loading shared libraries

### ISSUE THREAD

**Issue by WardCorn** [gsauthof/dracut-sshd#29]
Hi,

I'm trying dracut-sshd on Centos 8, and sshd will not start.
I have the logs : _ssh: error while loading shared libraries: libfipscheck.so.1_ (and the same with libutil after I fix fipscheck)

Maybe related the sshd host version ? I have the _OpenSSH_8.0p1, OpenSSL 1.1.1c FIPS  28 May 2019_

I fix it by adding the lib into the dracut module install script :
```
cp /usr/lib64/libfipscheck.so.1 $initdir/usr/lib64/libfipscheck.so.1
cp /usr/lib64/libfipscheck.so.1.2.1 $initdir/usr/lib64/libfipscheck.so.1.2.1
cp /usr/lib64/libutil.so.1 $initdir/usr/lib64/libutil.so.1
cp /usr/lib64/libutil-2.28.so $initdir/usr/lib64/libutil-2.28.so 
```
But how dracut can install dependencies automatically ?

**Comment by gsauthof** [comment]
Interesting, I tested this during RHEL 8 beta when the sshd didn't need those libraries.

Dracut also has a `inst_binary` function that is supposed to also install shared library dependencies.

Thus, ideally it should be enough to replace the

```
inst_simple /usr/sbin/sshd
```

call with:

```
inst_binary /usr/sbin/sshd
```

On the other hand when I do an `ldd` on the sshd on a Fedora system than I also get libfipscheck etc. as dependency. But in early-boot dracut environment it works without  them. Strange.

**Comment by WardCorn** [comment]
With _inst_binary_ it's work fine !

Strange there are no problem with Fedora.

**Comment by gsauthof** [comment]
Yesterday I checked a current Fedora 31 system (where the sshd does link against libfipscheck). I've just checked a current Fedora 32 system and there sshd doesn't link against libfipscheck.

So my current theory is that with Fedora it's an on-and-off thing. At some point sshd got this libfipscheck depedency but it was removed again at some later point.

In parallel, recent sshd versions on RHEL7 and 8 also have the libfipscheck dependency, while older RHEL 7 and RHEL 8 versions don't have this dependency. (cf. e.g. the CentOS 7 version I'm using as part of the dracut-sshd CI ...).

The sshd versions I've checked all depend on libutil but on the Fedora versions I've checked that dependency is already pulled in by some other dracut module.

Anyhow, using `inst_binary` for executables makes sense in general thus I'll replace the sshd install command as described in my last comment.
